### PR TITLE
#5140 fix count(null) for php 7.2+

### LINKED ISSUE #5140: CollectionType with php7.2
Environment
Php: Php-fpm7.2
Server: Nginx

#### Sonata packages

```
$ composer show --latest 'sonata-project/*'
sonata-project/admin-bundle              3.35.2 3.35.2 The missing Symfony A...
sonata-project/block-bundle              3.12.1 3.12.1 Symfony SonataBlockBu...
sonata-project/cache                     2.0.1  2.0.1  Cache library
sonata-project/core-bundle               3.10.0 3.10.0 Symfony SonataCoreBundle
sonata-project/datagrid-bundle           2.3.1  2.3.1  Symfony SonataDatagri...
sonata-project/doctrine-extensions       1.0.2  1.0.2  Doctrine2 behavioral ...
sonata-project/doctrine-orm-admin-bundle 3.6.1  3.6.1  Symfony Sonata / Inte...
sonata-project/easy-extends-bundle       2.5.0  2.5.0  Symfony SonataEasyExt...
sonata-project/exporter                  1.9.0  1.9.0  Lightweight Exporter ...
sonata-project/user-bundle               4.2.2  4.2.2  Symfony SonataUserBundle
```

#### Symfony packages

```
$ composer show --latest 'symfony/*'
symfony/asset                v4.1.1             v4.1.1            
symfony/browser-kit          v4.1.1             v4.1.1            
symfony/cache                v4.1.1             v4.1.1            
symfony/class-loader         v3.4.12            v3.4.12           
symfony/config               v4.1.1             v4.1.1            
symfony/console              v4.1.1             v4.1.1            
symfony/css-selector         v4.1.1             v4.1.1            
symfony/debug                v4.1.1             v4.1.1            
symfony/debug-bundle         v4.1.1             v4.1.1            
symfony/debug-pack           v1.0.5             v1.0.5            
symfony/dependency-injection v4.1.1             v4.1.1            
symfony/doctrine-bridge      v4.1.1             v4.1.1            
symfony/dom-crawler          v4.1.1             v4.1.1            
symfony/dotenv               v4.1.1             v4.1.1            
symfony/event-dispatcher     v4.1.1             v4.1.1            
symfony/expression-language  v4.1.1             v4.1.1            
symfony/filesystem           v4.1.1             v4.1.1            
symfony/finder               v4.1.1             v4.1.1            
symfony/flex                 v1.0.80            v1.0.80           
symfony/form                 v4.1.1             v4.1.1            
symfony/framework-bundle     v4.1.1             v4.1.1            
symfony/http-foundation      v4.1.1             v4.1.1            
symfony/http-kernel          v4.1.1             v4.1.1            
symfony/inflector            v4.1.1             v4.1.1            
symfony/intl                 v4.1.1             v4.1.1            
symfony/lts                  dev-master 6de50b2 dev-master 6de50b2
symfony/maker-bundle         v1.5.0             v1.5.0            
symfony/monolog-bridge       v4.1.1             v4.1.1            
symfony/monolog-bundle       v3.3.0             v3.3.0            
symfony/options-resolver     v4.1.1             v4.1.1            
symfony/orm-pack             v1.0.5             v1.0.5            
symfony/phpunit-bridge       v4.1.1             v4.1.1            
symfony/polyfill-ctype       v1.8.0             v1.8.0            
symfony/polyfill-intl-icu    v1.8.0             v1.8.0            
symfony/polyfill-mbstring    v1.8.0             v1.8.0            
symfony/polyfill-php72       v1.8.0             v1.8.0            
symfony/process              v4.1.1             v4.1.1            
symfony/profiler-pack        v1.0.3             v1.0.3            
symfony/property-access      v4.1.1             v4.1.1            
symfony/property-info        v4.1.1             v4.1.1            
symfony/routing              v4.1.1             v4.1.1            
symfony/security             v4.1.1             v4.1.1            
symfony/security-acl         v3.0.1             v3.0.1            
symfony/security-bundle      v4.1.1             v4.1.1            
symfony/serializer           v4.1.1             v4.1.1            
symfony/serializer-pack      v1.0.1             v1.0.1            
symfony/stopwatch            v4.1.1             v4.1.1            
symfony/swiftmailer-bundle   v3.2.2             v3.2.2            
symfony/templating           v4.1.1             v4.1.1            
symfony/test-pack            v1.0.2             v1.0.2            
symfony/translation          v4.1.1             v4.1.1            
symfony/twig-bridge          v4.1.1             v4.1.1            
symfony/twig-bundle          v4.1.1             v4.1.1            
symfony/validator            v4.1.1             v4.1.1            
symfony/var-dumper           v4.1.1             v4.1.1            
symfony/web-link             v4.1.1             v4.1.1            
symfony/web-profiler-bundle  v4.1.1             v4.1.1            
symfony/web-server-bundle    v4.1.1             v4.1.1            
symfony/webpack-encore-pack  v1.0.2             v1.0.2            
symfony/yaml                 v4.1.1             v4.1.1 
```

#### PHP version

```
$ php -v
PHP 7.2.4 (cli) (built: Apr  5 2018 00:37:24) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.2.4, Copyright (c) 1999-2018, by Zend Technologies
    with Xdebug v2.6.0, Copyright (c) 2002-2018, by Derick Rethans

```

## Subject

Try to using CollectionType w php7.2 but get some errors when using form

## Steps to reproduce

```
use Sonata\CoreBundle\Form\Type\CollectionType;

$formMapper
            ->add('items', CollectionType::class, [
                'type_options' => [
                    'delete' => false,
                    'delete_options' => [
                        'type'         => HiddenType::class,
                        'type_options' => [
                            'mapped'   => false,
                            'required' => false,
                        ]
                    ]
                ]
            ], [
                'edit' => 'inline',
                'inline' => 'table',
                'sortable' => 'position',
                'admin_code' => 'admin.dictionary_item'
            ]);
```
## Actual results

```
ErrorException:
Warning: count(): Parameter must be an array or an object that implements Countable

  at vendor/sonata-project/admin-bundle/src/Admin/AdminHelper.php:151
  at Sonata\AdminBundle\Admin\AdminHelper->appendFormFieldElement(object(DictionaryAdmin), null, 's5b380fd1ed622_items')
     (vendor/sonata-project/admin-bundle/src/Controller/HelperController.php:107)
  at Sonata\AdminBundle\Controller\HelperController->appendFormFieldElementAction(object(Request))
     (vendor/symfony/http-kernel/HttpKernel.php:149)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (vendor/symfony/http-kernel/HttpKernel.php:66)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
     (vendor/symfony/http-kernel/Kernel.php:188)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
     (public/index.php:37)
```

### PR DESCRIPTION
**Pull request by godzie44**
I am targeting this branch, because bug.

Closes #5140

## Changelog

```markdown
### Fixed

Worning "Parameter must be an array or an object that implements Countable" for count(null) in php 7.2
```

### CODE REVIEW

File: src/Admin/AdminHelper.php [hunk 1]
```diff
@@ -148,7 +148,7 @@ public function appendFormFieldElement(AdminInterface $admin, $subject, $element
                 $data[$childFormBuilder->getName()] = [];
             }
 
-            $objectCount = count($value);
+            $objectCount = null === $value ? 0 : count($value);
```
Comment (kunicmarko20) [inline src/Admin/AdminHelper.php]: what about `count($value ?? [])` but this is ok also.

Comment (kunicmarko20) [reply src/Admin/AdminHelper.php]: Ah, this is stable, ignore it, we still support php 5.6 :roll_eyes:

**Comment by greg0ire** [comment]
Thanks @godzie44 !

**Comment by godzie44** [comment]
You are welcome!

**Pull request merged**